internal/testutils/cgroup.go

```diff
@@ -0,0 +1,40 @@
+package testutils
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"golang.org/x/sys/unix"
+)
+
+func CreateCgroup(tb testing.TB) *os.File {
+	tb.Helper()
+
+	cgdir, err := ioutil.TempDir("/sys/fs/cgroup/unified", "ebpf-link")
```

Comment:
There is a fix in https://github.com/cilium/ebpf/pull/288/commits/94eca08ea7513611523023cfd6db0a84ce7fe6be maybe we can ask Zvi to create a separate PR for it.

Reply:
It would actually be helpful to export such a helper in the link package: This is boilerplate code that all users targeting cgroups will need.

Would it be acceptable to consider this as "not a regression" and "works when run with the test helper + in the CI" ?

Reply:
Fixed the onetime error issue with 94eca08 and made #352. I'm happy to turn this into an export if that's desired, but left it as a unit test function right now to keep scope minimized.

Reply:
I've just merged https://github.com/cilium/ebpf/pull/352, will give this a rebase.